test/e2e/daemon_set.go

```diff
@@ -155,11 +155,23 @@ var _ = framework.KubeDescribe("Daemon set [Serial]", func() {
 		err = checkDaemonStatus(f, dsName)
 		Expect(err).NotTo(HaveOccurred())
 
-		By("remove the node selector and wait for daemons to be unscheduled")
-		_, err = setDaemonSetNodeLabels(c, nodeList.Items[0].Name, map[string]string{})
+		By("Update the node label to green, and wait for daemons to be unscheduled")
+		nodeSelector[daemonsetColorLabel] = "green"
+		greenNode, err := setDaemonSetNodeLabels(c, nodeList.Items[0].Name, nodeSelector)
 		Expect(err).NotTo(HaveOccurred(), "error removing labels on node")
 		Expect(wait.Poll(dsRetryPeriod, dsRetryTimeout, checkRunningOnNoNodes(f, complexLabel))).
 			NotTo(HaveOccurred(), "error waiting for daemon pod to not be running on nodes")
+
+		By("Update DaemonSet node selector to green, and change its update strategy to RollingUpdate")
+		patch := fmt.Sprintf(`{"spec":{"template":{"spec":{"nodeSelector":{"%s":"%s"}}},"updateStrategy":{"type":"RollingUpdate"}}}`,
+			daemonsetColorLabel, greenNode.Labels[daemonsetColorLabel])
+		ds, err = c.Extensions().DaemonSets(ns).Patch(dsName, types.StrategicMergePatchType, []byte(patch))
```

Comment:
Not sure how to use the workaround like #42641 in this test, since controller will update DaemonSet as well and thus I'm not sure what resourceVersion I should wait on. I'll change this to retry on updates (instead of patch) until #42644 is fixed.